### ISSUE TITLE
stderr maxBuffer exceeded

### ISSUE THREAD

**Issue by davidcheung** [strongloop/loopback-sdk-angular-cli#47]
### Problem
CI failure due to maxBuffer exceeded

### Reason
when running `DEBUG=*` (which jenkins do) in a child_process
the child gets overwhelmed by globalization :(

### Possible solution
1. Exclude `strong-globalize`'s debug messages from Jenkins like we do for mocha
something like `DEBUG=*,-mocha,-strong-globalize` 

2. Increase maxBuffer size (for lb-ng tests), since Users are unlikely to generate their `lb-ng` with DEBUG=* it should solve this problem too

See http://ci.strongloop.com/job/loopback-sdk-angular-cli/label=ubuntu-4/2385/console

 ```coffeescript
1) generates "lbServices" module with app.restApiRoot url
Thu, 11 Aug 2016 21:20:16 GMT test --EXECFILE["/usr/local/bin/node" "/mnt/jenkins/workspace/loopback-sdk-angular-cli/e13f9a50/bin/lb-ng.js" "-m" "a-module-name" "/mnt/jenkins/workspace/loopback-sdk-angular-cli/e13f9a50/test/fixtures/app.js"]--
Thu, 11 Aug 2016 21:20:18 GMT test --FAILED--
Error: stderr maxBuffer exceeded
    at Socket.<anonymous> (child_process.js:285:14)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:153:18)
    at Socket.Readable.push (_stream_readable.js:111:10)
    at Pipe.onread (net.js:536:20)
---------------------------------------------
    at Object.exports.execFile (child_process.js:296:9)
    at exports.exec (child_process.js:111:18)
    at promisified (eval at makeNodePromisifiedEval (/mnt/jenkins/workspace/loopback-sdk-angular-cli/e13f9a50/node_modules/bluebird/js/main/promisify.js:198:12), <anonymous>:3:246)
    at runLbNg (/mnt/jenkins/workspace/loopback-sdk-angular-cli/e13f9a50/test/lb-ng.test.js:91:12)
    at Context.<anonymous> (/mnt/jenkins/workspace/loopback-sdk-angular-cli/e13f9a50/test/lb-ng.test.js:44:12)
    at callFn (/mnt/jenkins/workspace/loopback-sdk-angular-cli/e13f9a50/
```

**Comment by davidcheung** [comment]
so it appears that CI is failing for `angular-cli` is because of strong-globalize’s `debug msgs` http://ci.strongloop.com/job/loopback-sdk-angular-cli/2393/label=amazon-6/consoleFull
we spawn `child_process`, and amount of debug msg from strong-globalize is very high(when called from an outside module, because the loopback's globalize debug will go into the same `stderr`)
which causes stderr buffer to exceed.

**Comment by davidcheung** [comment]
https://github.com/strongloop/loopback-sdk-angular-cli/blob/6ffc51cb4c21ea47deb8d7afd431270979849df1/test/lb-ng.test.js#L63
this is also causing problem with tests, basically when i do 
`DEBUG=* mocha` then it would fail

```
  test --EXECFILE["/Users/davidcheung/.nvm/versions/node/v6.3.0/bin/node" "/Users/davidcheung/node-web/loopback-sdk-angular-cli/bin/lb-ng.js" "-m" "a-module" "/Users/davidcheung/node-web/loopback-sdk-angular-cli/test/fixtures/app.js" "/Users/davidcheung/node-web/loopback-sdk-angular-cli/test/sandbox/lb-services.js"]-- +0ms
  test --STDOUT--
Fri, 12 Aug 2016 20:01:43 GMT strong-globalize *** loadMsgFromFile(new res tag): loading: en messages.json
Fri, 12 Aug 2016 20:01:43 GMT strong-globalize 1 03bb755ed8213986b1862be937b296bb : The name for generated Angular module.
Fri, 12 Aug 2016 20:01:43 GMT strong-globalize 2 25eecfaa310e55ace4ef6eba593ba32e : Generating {0} for the API endpoint {1}
Fri, 12 Aug 2016 20:01:43 GMT strong-globalize 3 3669f5dd8a727a4f4247d7ad86993e0c : 
The code generator does not support applications based on
LoopBack versions older than 1.6.0. Please upgrade your project
to a recent version of LoopBack and run this tool again.

Fri, 12 Aug 2016 20:01:43 GMT strong-globalize 4 af39e85bbd74a37c9a8beec4268564bd : Generate Angular $resource services for your LoopBack application.
Usage:
    $0 [options] server/app.js [client/js/lb-services.js]
Fri, 12 Aug 2016 20:01:43 GMT strong-globalize 5 dbef1e1c2cc9f52ae9aacca33e9d92c2 : Saving the generated services source to {0}
Fri, 12 Aug 2016 20:01:43 GMT strong-globalize 6 dff77762368d4e45c34a01e661ad2c06 : Loading LoopBack app {0}
Fri, 12 Aug 2016 20:01:43 GMT strong-globalize 7 efc6bce92bcb061e3e354a888c7bd76a : Dumping to stdout
Fri, 12 Aug 2016 20:01:43 GMT strong-globalize 8 fc1c9755ca71e88d7aaa4be225d07bc8 : URL of the REST API end-point
Fri, 12 Aug 2016 20:01:44 GMT hahahaha OMG
Fri, 12 Aug 2016 20:01:44 GMT strong-globalize packMessage: Generate {{Angular $resource}} services for your {{LoopBack}} application.
Usage:
    $0 {{[options] server/app.js [client/js/lb-services.js]}} {
  "0": "Generate {{Angular $resource}} services for your {{LoopBack}} application.\nUsage:\n    $0 {{[options] server/app.js [client/js/lb-services.js]}}"
} undefined
Fri, 12 Aug 2016 20:01:44 GMT strong-globalize ~~~ lang = en af39e85bbd74a37c9a8beec4268564bd undefined /Users/davidcheung/node-web/loopback-sdk-angular-cli/node_modules/strong-globalize/lib/globalize.js
Fri, 12 Aug 2016 20:01:44 GMT strong-globalize packMessage: The name for generated {{Angular}} module. {
  "0": "The name for generated {{Angular}} module."
} undefined
Fri, 12 Aug 2016 20:01:44 GMT strong-globalize ~~~ lang = en 03bb755ed8213986b1862be937b296bb undefined /Users/davidcheung/node-web/loopback-sdk-angular-cli/node_modules/strong-globalize/lib/globalize.js
Fri, 12 Aug 2016 20:01:44 GMT strong-globalize packMessage: URL of the REST API end-point {
  "0": "URL of the REST API end-point"
} undefined
Fri, 12 Aug 2016 20:01:44 GMT strong-globalize ~~~ lang = en fc1c9755ca71e88d7aaa4be225d07bc8 undefined /Users/davidcheung/node-web/loopback-sdk-angular-cli/node_modules/strong-globalize/lib/globalize.js
Fri, 12 Aug 2016 20:01:44 GMT strong-globalize packMessage: Loading {{LoopBack}} app %j {
  "0": "Loading {{LoopBack}} app %j",
  "1": "/Users/davidcheung/node-web/loopback-sdk-angular-cli/test/fixtures/app.js"
} [ '"/Users/davidcheung/node-web/loopback-sdk-angular-cli/test/fixtures/app.js"' ]
Fri, 12 Aug 2016 20:01:44 GMT strong-globalize ~~~ lang = en dff77762368d4e45c34a01e661ad2c06 ["\"/Users/davidcheung/node-web/loopback-sdk-angular-cli/test/fixtures/app.js\""] /Users/davidcheung/node-web/loopback-sdk-angular-cli/node_modules/strong-globalize/lib/globalize.js
Fri, 12 Aug 2016 20:01:44 GMT strong-globalize packMessage: Generating %j for the API endpoint %j {
  "0": "Generating %j for the API endpoint %j",
  "1": "a-module",
  "2": "/rest-api-root"
} [ '"a-module"', '"/rest-api-root"' ]
Fri, 12 Aug 2016 20:01:44 GMT strong-globalize ~~~ lang = en 25eecfaa310e55ace4ef6eba593ba32e ["\"a-module\"","\"/rest-api-root\""] /Users/davidcheung/node-web/loopback-sdk-angular-cli/node_modules/strong-globalize/lib/globalize.js
Fri, 12 Aug 2016 20:01:44 GMT strong-globalize packMessage: Saving the generated services source to %j {
  "0": "Saving the generated services source to %j",
  "1": "/Users/davidcheung/node-web/loopback-sdk-angular-cli/test/sandbox/lb-services.js"
} [ '"/Users/davidcheung/node-web/loopback-sdk-angular-cli/test/sandbox/lb-services.js"' ]
Fri, 12 Aug 2016 20:01:44 GMT strong-globalize ~~~ lang = en dbef1e1c2cc9f52ae9aacca33e9d92c2 ["\"/Users/davidcheung/node-web/loopback-sdk-angular-cli/test/sandbox/lb-services.js\""] /Users/davidcheung/node-web/loopback-sdk-angular-cli/node_modules/strong-globalize/lib/globalize.js

--STDERR--
Loading LoopBack app "/Users/davidcheung/node-web/loopback-sdk-angular-cli/test/fixtures/app.js"
Fri, 12 Aug 2016 20:01:44 GMT express:application set "x-powered-by" to true
Fri, 12 Aug 2016 20:01:44 GMT express:application set "etag" to 'weak'
Fri, 12 Aug 2016 20:01:44 GMT express:application set "etag fn" to [Function: wetag]
Fri, 12 Aug 2016 20:01:44 GMT express:application set "env" to 'development'
Fri, 12 Aug 2016 20:01:44 GMT express:application set "query parser" to 'extended'
Fri, 12 Aug 2016 20:01:44 GMT express:application set "query parser fn" to [Function: parseExtendedQueryString]
Fri, 12 Aug 2016 20:01:44 GMT express:application set "subdomain offset" to 2
Fri, 12 Aug 2016 20:01:44 GMT express:application set "trust proxy" to false
Fri, 12 Aug 2016 20:01:44 GMT express:application set "trust proxy fn" to [Function: trustNone]
Fri, 12 Aug 2016 20:01:44 GMT express:application booting in development mode
Fri, 12 Aug 2016 20:01:44 GMT express:application set "view" to [Function: View]
Fri, 12 Aug 2016 20:01:44 GMT express:application set "views" to '/Users/davidcheung/node-web/loopback-sdk-angular-cli/views'
Fri, 12 Aug 2016 20:01:44 GMT express:application set "jsonp callback name" to 'callback'
Fri, 12 Aug 2016 20:01:44 GMT loopback:datasource Settings: {"debug":true}
Fri, 12 Aug 2016 20:01:44 GMT loopback:registry Attached model `User` to dataSource `Memory`
Fri, 12 Aug 2016 20:01:44 GMT express:application set "restApiRoot" to '/rest-api-root'
Generating "a-module" for the API endpoint "/rest-api-root"
Fri, 12 Aug 2016 20:01:44 GMT express:router use / <anonymous>
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter remoting options: {}
Fri, 12 Aug 2016 20:01:44 GMT express:router use / <anonymous>
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT express:router use / urlencodedParser
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT express:router use / jsonParser
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter registering REST handler for class "User"
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter     method User.create
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter     method User.upsert
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter     method User.exists
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter     method User.findById
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter     method User.find
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter     method User.findOne
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter     method User.updateAll
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter     method User.deleteById
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter     method User.count
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter     method User.prototype.updateAttributes
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter     method User.createChangeStream
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter         put /:id restPrototypeMethodHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:route new /:id
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /:id
Fri, 12 Aug 2016 20:01:44 GMT express:router:route put /:id
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter         put / restStaticMethodHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:route new /
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT express:router:route put /
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter         post /change-stream restStaticMethodHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:route new /change-stream
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /change-stream
Fri, 12 Aug 2016 20:01:44 GMT express:router:route post /change-stream
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter         post /update restStaticMethodHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:route new /update
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /update
Fri, 12 Aug 2016 20:01:44 GMT express:router:route post /update
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter         post / restStaticMethodHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:route new /
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT express:router:route post /
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter         head /:id restStaticMethodHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:route new /:id
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /:id
Fri, 12 Aug 2016 20:01:44 GMT express:router:route head /:id
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter         get /change-stream restStaticMethodHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:route new /change-stream
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /change-stream
Fri, 12 Aug 2016 20:01:44 GMT express:router:route get /change-stream
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter         get /count restStaticMethodHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:route new /count
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /count
Fri, 12 Aug 2016 20:01:44 GMT express:router:route get /count
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter         get /findOne restStaticMethodHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:route new /findOne
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /findOne
Fri, 12 Aug 2016 20:01:44 GMT express:router:route get /findOne
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter         get /:id/exists restStaticMethodHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:route new /:id/exists
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /:id/exists
Fri, 12 Aug 2016 20:01:44 GMT express:router:route get /:id/exists
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter         get /:id restStaticMethodHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:route new /:id
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /:id
Fri, 12 Aug 2016 20:01:44 GMT express:router:route get /:id
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter         get / restStaticMethodHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:route new /
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT express:router:route get /
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter         del /:id restStaticMethodHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:route new /:id
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /:id
Fri, 12 Aug 2016 20:01:44 GMT express:router:route delete /:id
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT express:router use / restRemoteMethodNotFound
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT strong-remoting:rest-adapter     at /Users
Fri, 12 Aug 2016 20:01:44 GMT express:router use /Users router
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /Users
Fri, 12 Aug 2016 20:01:44 GMT express:router use / restUrlNotFound
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Fri, 12 Aug 2016 20:01:44 GMT express:router use / restErrorHandler
Fri, 12 Aug 2016 20:01:44 GMT express:router:layer new /
Saving the generated services source to "/Users/davidcheung/node-web/loopback-sdk-angular-cli/test/sandbox/lb-services.js"

```

**Comment by davidcheung** [comment]
@rmg @bajtos @Setogit any thoughts on how to combat this.

**Comment by bajtos** [comment]
I would personally exclude `DEBUG` env var from the environment provided to the child processes.

**Comment by davidcheung** [comment]
fixed in #50

**Comment by rmg** [comment]
CI only sets `DEBUG` in the second run of the tests, which means they are failing for some other reason before `DEBUG` is set. I suspect this PR would have passed with just the `--timeout` change by itself.
